compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -296,8 +303,24 @@ BLangStatementExpression desugar(BLangQueryExpr queryExpr, SymbolEnv env,
             BType refType = Types.getReferredType(queryExpr.getBType());
             if (TypeTags.isXMLTypeTag(refType.tag) || (refType.tag == TypeTags.UNION
                     && ((BUnionType) refType).getMemberTypes().stream().allMatch(memType ->
-                                    TypeTags.isXMLTypeTag(Types.getReferredType(memType).tag)))) {
-                result = getStreamFunctionVariableRef(queryBlock, QUERY_TO_XML_FUNCTION, Lists.of(streamRef), pos);
+                    TypeTags.isXMLTypeTag(Types.getReferredType(memType).tag == TypeTags.INTERSECTION ?
+                            ((BIntersectionType) Types.getReferredType(memType)).effectiveType.tag :
+                            Types.getReferredType(memType).tag)))) {
+                BLangLiteral isReadonly = ASTBuilderUtil.createLiteral(null, symTable.booleanType, false);
+                if (Symbols.isFlagOn(refType.flags, Flags.READONLY)) {
+                    isReadonly.value = true;
+                } else if (refType.tag == TypeTags.UNION) {
+                    for (BType type : ((BUnionType) refType).getMemberTypes()) {
+                        if (Symbols.isFlagOn(type.flags, Flags.READONLY)) {
+                            isReadonly.value = true;
+                            continue;
```

Comment:
Codecov warnings?

Reply:
will check.

Reply:
I debugged and checked. Seems it was a false warning, because this if statement is covered with tests.

Reply:
Line L317 cannot be covered due to https://github.com/ballerina-platform/ballerina-lang/issues/36476.